### ISSUE TITLE
robustness of optimization

### ISSUE THREAD

**Issue by hughsalimbeni** [GPflow/GPflow#119]
Optimization using the standard numpy optimizers seems rather fragile. e.g.
 
```
N = 40
X = np.reshape(np.random.uniform(0, 1, N), (N, 1))
Y = np.sin(3*X) + np.reshape(np.random.randn(np.size(X)), X.shape)*0.001

model_GPy = GPy.models.GPRegression(X, Y, GPy.kern.RBF(1))
model_GPy.optimize() # works fine 

model_GPflow = GPflow.gpr.GPR(X, Y, GPflow.kernels.RBF(1))
model_GPflow.optimize(method='L-BFGS-B') # fails on 6/10 attempts 
model_GPflow.optimize(method='CG') #  fails on 9/10 attempts 
model_GPflow.optimize(method='BFGS') # fails on 10/10 attempts 
```
However, using  another optimizer: 
```
step = 0.5
model_GPflow.optimize(method=tf.train.AdagradOptimizer(step), max_iters=1000) # works
```
But if `step` is less than 0.1 it doesn't get there after 1000 iterations, and if `step` is greater than about 1 it doesn't seem to find the right answer.

Clearly setting the initial likelihood variance to something smaller solves this particular example, but in general it seems problematic if the out-the-box optimizer fails with misspecified hyperparameters, even in simple cases. Could GPflow be made more robust to these numerical issues?

**Comment by jameshensman** [comment]
@alexisboukouvalas does this help at all with your numerical issues?

**Comment by jameshensman** [comment]
This is fixed in #120
